### ISSUE TITLE
[Nuke] Cryptomatte Gizmo - RemoveChannels

### ISSUE THREAD

**Issue by dgirondi** [Psyop/Cryptomatte#48]
Hi,

Just wondering if when this is checked the Cryptomatte Metadata should still be available for downstream nodes if the crypto layers are not being kept.

Adding a ModifyMetaData node below the Remove node with the following "action: remove, key: exr/cryptomatte/*" could address this and keep things even cleaner downstream.

Cheers,
Diogo

**Comment by jonahfriedman** [comment]
Agreed, this makes sense. Pull requests welcome. The remove channels knob is something that we probably would have removed in the last round, but that would have broken compatibility. Since it is there though, this is a logical extension of it. 

If anyone addresses this, some other things to think about: 
- the metadata may also take the form of cryptomatte/*, if the read node has "do not attach prefix" set, and the metadata should be stripped there too
- there's a test in the suite for remove channels, it'd be good to add this to that test
- Decryptomatte should probably also remove the metadata if remove channels does. (I'd be ok acceping a pull request without this one, as it has a bad "does anyone care" to complexity ratio, so we can live without it. 

Also a question: Do you find having the remove channels knob useful?

**Comment by dgirondi** [comment]
Personally I don't. Not too fond of nodes for one thing that end up doing another and don't display that on the DAG. 

If it was for me I would remove that from that node and add another Gizmo for it. A "CryptoRemove" that would remove the Cryptomattes and accompanying metadata exclusively.

**Comment by jonahfriedman** [comment]
Yeah, I agree. If this were a new project I'd definitely remove it.

**Comment by dgirondi** [comment]
I'll fork the project to see if I can contribute with something other than being a pita ;)

Thanks Jonah!

**Comment by jonahfriedman** [comment]
Haha, no problem, looking forward to it! And it's no problem, all your reported issues are well thought out. You've obviously spent time with the toolset and I'm very happy to be having this discussion.

**Comment by dgirondi** [comment]
Using it on a daily basis here in every single project, it took me a few months to convince people to jump in but now there is no turning back.

We are only using vray multimattes for scenarios where refraction/reflection are required.

**Comment by dgirondi** [comment]
By the way, since I've mentioned it will ever be possible for Cryptomatte to support refracted/reflected mattes?

That would be the cherry on the cake.

**Comment by jonahfriedman** [comment]
Hi @dgirondi,

It is possible to support refracted/reflected mattes, and this is largely a task for the renderer. However, I don't think there are any publicly available implementations which actually do that.

**Comment by vkoylazov** [comment]
We've done some work on that for the next version of V-Ray, at least as far as refraction is concerned (objects visible behind glass etc).

Best regards,
Vlado

**Comment by dgirondi** [comment]
That is awesome @Vlado!

We are currently using a mix of Cryptomatte and Multimattes in VRay but
getting rid of multimattes entirely will be great.


Cheers,
Diogo

**Comment by fredrikaverpil** [comment]
@vkoylazov shows up, like usual, with great news. Love it!
